lib/HLSDownloader.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Created by AsherS on 8/20/15.
+ */
+
+/**
+ * Created by AsherS on 8/11/15.
+ */
+
+var Q = require('q');
+var _ = require('underscore');
+var path = require('path');
+var url = require('url');
+var downloadUtils = require('./DownloadUtils');
+var fileUtils = require('./FileUtils');
+var hlsUtils = require('./HLSUtils');
+
+var HLSDownloader = (function () {
+
+    function HLSDownloader(m3u8Url, destPath) {
+        this.streamUrl = m3u8Url;
+        this.destPath = destPath;
+    }
+
+    var downloadIndefinitely = function (url, dest) {
+        downloadPlaylist(url, dest)
+            .then(function(targetDuration){
+                console.log('results: ' + targetDuration);
+            })
+            .catch(function(err){
+                console.log('error ' + err);
+            })
+            .done();
```

Comment:
.done() has no meaning if you .catch beforehand.
Also, the polling interval should be configurable (as you mentioned)